### PR TITLE
Tests for conserving moisture

### PR DESCRIPTION
**Pull request by bischtob**
This adds regression and validation tests for the moist equilibirum runs.

### CODE REVIEW

**Comment by bischtob** [comment]
bors try

File: src/Models/Nonhydrostatic3DModels/equations_moisture.jl [hunk 1]
```diff
@@ -55,12 +56,12 @@
        Geometry.Covariant123Vector(interp_f2c(w))
 
     # hyperdiffusion
-    dρq_tot = 0.0 .* dρq_tot
+    #dρq_tot = FT(0) .* dρq_tot
     χq = @. hwdiv(hgrad(ρq_tot / ρ))
     Spaces.weighted_dss!(dρq_tot)
     @. dρq_tot = -κ₄ * hwdiv(ρ * hgrad(χq))
```
Comment (akshaysridhar) [inline src/Models/Nonhydrostatic3DModels/equations_moisture.jl:62]: If the working fluid is moist air then the diffusive / hyperdiffusive moisture terms should appear in the tendency equation for \rho as well.

Comment (bischtob) [reply src/Models/Nonhydrostatic3DModels/equations_moisture.jl:62]: Not sure, I understand :).

Comment (jiahe23) [inline src/Models/Nonhydrostatic3DModels/equations_moisture.jl:62]:
```suggestion
    @. dρq_tot = -κ₄ * hwdiv(ρ * hgrad(χq))
    @. dρ = dρq_tot
```

Comment (jiahe23) [reply src/Models/Nonhydrostatic3DModels/equations_moisture.jl:62]: @akshaysridhar  Will this work?

File: src/Models/Nonhydrostatic3DModels/equations_pressure.jl [hunk 2]
```diff
@@ -69,17 +69,20 @@ end
     uvw = @. Geometry.Covariant123Vector(uh) +
        Geometry.Covariant123Vector(interp_f2c(w))
     Φ = calculate_gravitational_potential(Y, Ya, params, FT)
-
     e_int = @. ρe_tot / ρ - Φ - norm(uvw)^2 / 2
     q_tot = @. ρq_tot / ρ
+
+    # saturation adjustment
+    maxiter = 1
+    temperature_tol = FT(0.1)
```
Comment (charleskawczynski) [inline src/Models/Nonhydrostatic3DModels/equations_pressure.jl:77]: Why aren't we using the default parameters?

File: src/Models/Nonhydrostatic3DModels/equations_moisture.jl (hunk 1 above)
Comment (bischtob) [reply src/Models/Nonhydrostatic3DModels/equations_moisture.jl:62]: For now, we are going to ignore the effect :)

File: src/Models/Nonhydrostatic3DModels/equations_pressure.jl (hunk 2 above)
Comment (bischtob) [reply src/Models/Nonhydrostatic3DModels/equations_pressure.jl:77]: From what I remember @tapios has wanted us to use a single Newton iteration.

File: src/Models/Nonhydrostatic3DModels/equations_moisture.jl (hunk 1 above)
Comment (bischtob) [reply src/Models/Nonhydrostatic3DModels/equations_moisture.jl:62]: Yes, that would work

**Comment by bischtob** [comment]
bors try

**Pull request merged**